shared/chat/manage-channels/container.js

```diff
@@ -88,27 +96,21 @@ const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, routePath, routePro
       oldChannelState: ChannelMembershipState,
       nextChannelState: ChannelMembershipState
     ) => {
-      const channelsToChange = pickBy(
-        nextChannelState,
-        (inChannel: boolean, channelname: string) => inChannel !== oldChannelState[channelname]
-      )
-      dispatch(TeamsGen.createSaveChannelMembership({teamname, channelState: channelsToChange}))
+      dispatch(createSaveChannelMembership(teamname, oldChannelState, nextChannelState))
+      dispatch(navigateUp())
     },
-    _onPreview: (conversationIDKey: ChatTypes.ConversationIDKey, previousPath?: string[]) => {
-      dispatch(Chat2Gen.createSelectConversation({conversationIDKey, reason: 'preview'}))
+    _onView: (
+      oldChannelState: ChannelMembershipState,
+      nextChannelState: ChannelMembershipState,
+      channel: string,
+      conversationIDKey: ChatTypes.ConversationIDKey
+    ) => {
+      const selected = nextChannelState[channel]
+      dispatch(createSaveChannelMembership(teamname, oldChannelState, nextChannelState))
```

Comment:
we dispatch a save here even when normally it would be impossible to save (i.e., when the getChannels RPC is still in progress). From the way that the channel states are changed, it seems ok to me, though.